client/add/add.ctrl.js

```diff
@@ -123,10 +123,27 @@ module.exports = class Add {
            toastAction,
            toastAction ? `/pokemon/${successfulUploads[0].created.id}` : undefined);
       })
-      .filter(line => line.success && line.created.box === box.data.id)
-      .then(lines => lines.slice(0, BOX_PAGE_SIZE - (box.data.contents.length % BOX_PAGE_SIZE)))
+      .filter(line => line.success)
       .map(line => line.created)
-      .then(lines => box.data.contents.push(...lines))
+      .reduce((overall, line) => {
+        if (overall[line.box]) {
+          overall[line.box].push(line);
+        } else {
+          overall[line.box] = [line];
+        }
+        return overall;
+      }, {})
+      .then(lines => {
+        Object.keys(lines).forEach(
+          (line) => line.slice(0, BOX_PAGE_SIZE - (box.data.contents.length % BOX_PAGE_SIZE))
+        );
+        return lines;
```

Comment:
This becomes a lot more complex to do if we change to using a map of id to box, because this map doesn't exist anymore.